frontside/lib/utils/debouncer.dart

```diff
@@ -0,0 +1,17 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+class Debouncer {
+  Duration delay;
+  Timer? _timer;
+
+  Debouncer({this.delay = const Duration(milliseconds: 500)});
+
+  void debounce(VoidCallback callback) {
+    if (_timer != null && _timer!.isActive) {
```

Comment:
I don't get it why I have to use `!` in this line even though I checked if the variable is null.

Reply:
because dart will not stop of first one but it will look at second condition as well in very situation best way I know is to change it to _timer?.isActive ?? false that will check if is null and when it is it will return null not call method and wen is null ?? operator return what we set on right side

Reply:
728028b